meeshkan/serve/mock/response_matcher.py

```diff
@@ -46,136 +39,37 @@ def default_response(self, msg):
             timestamp=None,
         )
 
+    def _match_response(self, request, spec, storage):
+        try:
+            return MeeshkanFaker(self._fkr, request, spec, storage).execute()
+        except FakerException as e:
+            return self.match_error(str(e), request)
+
     def get_response(self, request: Request) -> Response:
-        # TODO: tight coupling here
-        # try to decouple...
-        specs = rest_middleware_manager.spew(request, self._specs)
-        matches = match_request_to_openapi(request, specs)
-        if len(matches) == 0:
-            return self.match_error(
-                "Could not find an OpenAPI schema for the host %s." % request.host,
-                request,
+        if request.method.value is None:
+            method_error = "Could not find a method %s for path %s on hostname %s." % (
+                request.method.value,
+                request.path,
+                request.host,
             )
-        random.shuffle(matches)
-        first_match = matches[0]
-        path_error = "Could not find a path %s on hostname %s." % (
-            request.path,
-            request.host,
-        )
-        method_error = "Could not find a method %s for path %s on hostname %s." % (
-            request.method.value,
-            request.path,
-            request.host,
-        )
-        if first_match.api.paths is None:
-            return self.match_error(path_error, request)
-        if len(first_match.api.paths.items()) == 0:
-            return self.match_error(path_error, request)
-        path_candidates = [x for x in first_match.api.paths.values()]
-        random.shuffle(path_candidates)
-        path_candidate = path_candidates[0]
+            return self.match_error(method_error, request)
 
-        method = {
-            "get": path_candidate.get,
-            "post": path_candidate.post,
-            "put": path_candidate.put,
-            "delete": path_candidate.delete,
-            "options": path_candidate.options,
-            "head": path_candidate.head,
-            "patch": path_candidate.patch,
-            "trace": path_candidate.trace,
-        }.get(str(request.method.value), None)
+        schemas = rest_middleware_manager.spew(request, self._specs)
+        match = match_request_to_openapi(request, schemas)
```

Comment:
There will be conflicts to master here as this is missing the matching to security scheme.